### PR TITLE
release: add a few performance-related notes

### PR DESCRIPTION
**Pull request by theuni**
Tried to keep these simple and human-readable.

I'm sure I've left out a few, please chime in with any others.

Because these all depend on specific use-cases (number of cores, peers, bandwidth, etc), I've left out any specific performance figures. If preferred, we could instead list numbers plus reference system/scenario.

### CODE REVIEW

File: doc/release-notes.md [hunk 1]
```diff
@@ -33,6 +33,25 @@ frequently tested on them.
...
 ===============
 
+Performance Improvements
+--------------
+
+Validation speed and network propogation performance have been greatly
+improved, leading to much shorter sync and initial block download times.
+
+- The script signature cache has been reimplemented as a high-performance,
+  low-overhead "cuckoo cache"
```
Comment (TheBlueMatt) [inline doc/release-notes.md]: ...", allowing for more signatures to be cached and cache lookups to be performed much faster."

File: doc/release-notes.md [hunk 2]
```diff
@@ -33,6 +33,25 @@ frequently tested on them.
...
+- Assumed-valid blocks have been introduced which allows script validation to
+  be skipped for ancestors of known-good blocks, without changing the security
+  model. See below for more details.
+- In some cases, compact blocks are now relayed before being fully validated as
+  per BIP152.
```
Comment (TheBlueMatt) [inline doc/release-notes.md]: ...", often allowing relay to happen fully without waiting for block validation."

**Review by TheBlueMatt — COMMENTED** [review]
This should definitely make it in for release time, if we can.

**Comment by MarcoFalke** [comment]
Could mention #8610 as well?

File: doc/release-notes.md [hunk 3]
```diff
@@ -33,6 +33,25 @@ frequently tested on them.
...
+- P2P networking has been refactored with a focus on concurrency and
+  throughput. Network operations are no longer bottlenecked by validation. As a
+  result, block fetching is several times faster that previous releases in many
```
Comment (jnewbery) [inline doc/release-notes.md]: ...several times faster **than** previous...

File: doc/release-notes.md (hunk 2 above)
Comment (theuni) [inline doc/release-notes.md]: I'm not sure what you're getting at here, sounds the same to me :)
Please suggest a complete alternative.

**Comment by theuni** [comment]
Updated with the suggestions above, except for the one I'm unclear about.

File: doc/release-notes.md [hunk 4]
```diff
@@ -33,6 +33,28 @@ frequently tested on them.
 Notable changes
 ===============
 
+Performance Improvements
+--------------
+
+Validation speed and network propogation performance have been greatly
```
Comment (sdaftuar) [inline doc/release-notes.md]: "propagation"

File: doc/release-notes.md (hunk 2 above)
Comment (TheBlueMatt) [inline doc/release-notes.md]: Mmm, nevermind, you're probably right.

File: doc/release-notes.md [hunk 5]
```diff
@@ -33,6 +33,28 @@ frequently tested on them.
...
+improved, leading to much shorter sync and initial block download times.
```
Comment (sipa) [inline doc/release-notes.md]: Perhaps qualify this with "in some configurations"... people who are bandwidth starved or I/O bottlenecked won't need much difference.

**Review by jtimon — APPROVED** [review]

**Pull request merged**